### PR TITLE
Update change-log.md

### PR DESCRIPTION
**Pull request by MartaD**

### CODE REVIEW

File: docs/change-log.md [hunk 1]
```diff
@@ -2,6 +2,21 @@
 
 This document provides an overview of the changes we are making over time. Feel free to reach out to us in case of any specific questions.
 
+## Product Update March 22nd, 2024
+
+- **Improved** The `artefact` trigger in Humanitec Pipelines now enforces serial execution, so a batch will be kept open until the previous batch has finished running. No explicit concurrency-group setting is required.
```
Comment (MartaD) [inline docs/change-log.md:7]: @astromechza please let me know if this is user-facing improvement or an implementation one, or how else I could describe that update in a way users would understand, like pipelines running in parallel without explicit configuration?

File: docs/change-log.md [hunk 2]
```diff
@@ -2,6 +2,21 @@
...
+
+- **Improved** Terraform Driver now provides a more meaningful message when the [service account](https://developer.humanitec.com/integration-and-extensions/drivers/generic-drivers/terraform/#runner-object) specified in the external runner configuration does not exist in the target namespace.
+  
+- **Fixed:** Minor [RBAC](https://developer.humanitec.com/platform-orchestrator/security/rbac) issue that hid the option to create new applications for users with the Manager role.
```
Comment (MartaD) [inline docs/change-log.md:11]:
```suggestion
- **Fixed:** Minor [RBAC](https://developer.humanitec.com/platform-orchestrator/security/rbac) issue that hid the option to create new applications in UI for users with the Manager role.
```

File: docs/change-log.md [hunk 3]
```diff
@@ -2,6 +2,21 @@
...
+## Product Update March 22nd, 2024
+
+- **Improved** The `artefact` trigger in Humanitec Pipelines now enforces serial execution, so a batch will be kept open until the previous batch has finished running. No explicit concurrency-group setting is required.
+
+- **Improved** Terraform Driver now provides a more meaningful message when the [service account](https://developer.humanitec.com/integration-and-extensions/drivers/generic-drivers/terraform/#runner-object) specified in the external runner configuration does not exist in the target namespace.
+  
+- **Fixed:** Minor [RBAC](https://developer.humanitec.com/platform-orchestrator/security/rbac) issue that hid the option to create new applications in UI for users with the Manager role.
+
+- **[Humanitec CLI](https://developer.humanitec.com/platform-orchestrator/cli):** Version [0.18.0](https://github.com/humanitec/cli/releases/tag/v0.18.0) is now available. Changes since the last announcement:
+  - `0.15.0` adding the possibility to filter active resource usage by deployment.
```
Comment (chrishumanitec) [inline docs/change-log.md:14]: The order of the versions seems a bit weird? Should be descending

**Review by chrishumanitec — CHANGES_REQUESTED** [review]

File: docs/change-log.md (hunk 3 above)
Comment (MartaD) [reply docs/change-log.md]: done

**Comment by MartaD** [comment]
@chrishumanitec feel free to add any more comments, I'll fix post-merge

**Pull request merged**